### DISCUSSION TITLE
Progress bar set doesn't work in loop

### DISCUSSION THREAD

**Discussion by Azuremoon13** [TomSchimansky/CustomTkinter#265] [General]
progressbar.set(value) doesnt work in a loop, only sets the last value in the list

**Comment by louislautz** [comment]
@Azuremoon13 I am working on a project right now where I do exactly that.
Note that you may need to do this in a separate threat. If you dont, the loop will finish running and only then the widget will be updated. Basically the app freezes until the loop is done. By creating a new thread, the app can continue as usual while also running the loop in parallel.

I put my loop in a new thread with the "threading" library in python and it works just fine :D

Hope that makes sense :D

**Comment by TomSchimansky** [comment]
In tkinter there is the .after() method to do this, to avoid using separate threads, which can cause problems in tkinter:

```python
import customtkinter


class App(customtkinter.CTk):
    def __init__(self):
        super().__init__()

        self.value = 0
        self.progressbar = customtkinter.CTkProgressBar(self)
        self.progressbar.pack(pady=20)
        self.progressbar.set(self.value)

        self.update_progressbar()

    def update_progressbar(self):
        self.value += 0.01
        self.progressbar.set(self.value)

        if self.value < 1:
            self.after(20, self.update_progressbar)  # call update_progressbar after 20 ms


app = App()
app.mainloop()
```

**Reply by Azuremoon13** [reply]
oo brill cheers! didnt even think about the .after()

**Reply by Azuremoon13** [reply]
is there a way to block anything else from running while its update the progressbar?
